### DISCUSSION TITLE
兼容lua社区涉及到网络的开源库

### DISCUSSION THREAD

**Discussion by Tianze-Chen** [cloudwu/skynet#1209] [Ideas]
很多现成的lua工具，一旦涉及到socket，都无法在skynet上正常运行，本质上造成了skynet社区和lua社区的割裂。
例如rabbitmq-lua是无法在skynet上运行的，必须有一个rabbitmq-skynet才可以。
而本身skynet支持越来越多的功能，维护成本也会增加，导致对一些用的人比较少，但是又很重要的模块无法官方支持。
如果可以直接拿外部的库来跑，或者skynet社区维护的库可以被其他使用lua的社区使用，感觉会是一件比较好的事情。

**Comment by hanxi** [comment]
跟 openresty 是一个道理，socket 不一样，相关的库还是需要重新实现的，可以像 openresty 一样维护一些常用第三方库，不包含到 skynet 中的。好像 wiki 中已经有人这么干了。

**Comment by hanxi** [comment]
参考这里 https://github.com/cloudwu/skynet/wiki/ExternalService
